### PR TITLE
fix(desktop): name the session in the usage activity task column

### LINKED ISSUE #4218: fix(desktop): Usage Statistics activity log always shows "未命名会话" in the 任务 (Task) column
## What happened

On the Usage Statistics page (使用统计) → Activity records (活动记录), the **任务 (Task)** column shows `未命名会话 · <short id>` ("Untitled session · …") for **every** row — even when the underlying session has a real title in the WorkHub.

Expected: named sessions show their real title in the Task column; only genuinely untitled sessions fall back to `未命名会话 · <id>`.

**This is a regression.** #3833 fixed exactly this column by carrying the session name (`SessionHeader.name`) through the old `packages/storage/src/usage-stats-store.ts` pipeline. That pipeline was later replaced by the Runtime Host usage projection (#3697), and the old store was deleted (#4060). The new pipeline only carries `sessionId`, never the session name, so `UsageRequestLog.sessionName` is declared but never populated — and the UI's `usageSessionDisplayLabel` always hits the untitled fallback for every row.

## How to reproduce

1. Open Settings → Usage Statistics (使用统计).
2. Have some model/tool activity in sessions that carry real titles.
3. Open the Activity records (活动记录) tab and look at the 任务 (Task) column.
4. Every row reads `未命名会话 · <short id>`, regardless of whether the session actually has a name.

## Environment

- Maka commit: `aa586237e`
- OS and version: macOS (Darwin 24.6.0)
- Surface: Desktop
- Node.js version: running from source

## Logs, screenshots, or additional context

Root cause — `sessionName` is dropped at every layer of the current usage pipeline:

- `packages/core/src/settings.ts` — `UsageRequestLog.sessionName` is declared (with a doc comment) but no producer fills it.
- `apps/desktop/src/main/runtime-host-usage-ipc-main.ts` — `projectLlmLog` / `projectToolLog` carry `sessionId`/`turnId` but never `sessionName`.
- `packages/runtime-host/src/protocol/usage-pricing.ts` — the projection types (`LlmUsageLogProjection` / `ToolUsageLogProjection`) have no session-name field at all.
- The UI already reads `row.sessionName` (`apps/desktop/src/renderer/settings/usage-settings-page.tsx`, `usageSessionDisplayLabel`), so no UI change is needed — the data layer just needs to populate the field again.

Raw usage rows (`UsageLogRow`) are keyed only by `sessionId`; the human-readable title lives in the session catalog. Recovering the name is a `sessionId → name` join, which the deleted store used to do.

### PR DESCRIPTION
**Pull request by liuxiaocs7**
## Summary

On the Usage Statistics page (使用统计) → Activity records (活动记录), the **任务 (Task)** column showed `未命名会话 · <short id>` ("Untitled session") for **every** row — even sessions that have a real title.

This is a regression. #3833 fixed exactly this column by carrying the session name (`SessionHeader.name`) through the old `packages/storage/src/usage-stats-store.ts` pipeline. #3697 then moved usage onto the Runtime Host projection and #4060 deleted that store. The new pipeline only carried `sessionId`, never the title, so `UsageRequestLog.sessionName` was declared-but-never-populated and the UI's `usageSessionDisplayLabel` always fell back to the untitled label.

Resolve the title **on the Host**, where every usage-bearing session is reachable:

- `HostUsagePricingCoordinator` takes a session-title reader (`stores.sessionStore.readHeaderSnapshot`). Per logs page it resolves the title for each row's `sessionId` and emits it as a new bounded `sessionTitle` field on the LLM/tool usage projections (`projectText`, ≤1024 bytes).
- Reading the **durable session header by id** bypasses the catalog's `role`/`preparing`/`ledger-v0` filters (`sqliteOrdinarySessionRolePredicate` et al.), so reserved-role, coordination (e.g. WorkHub), and legacy sessions are named too — not just ordinary ones.
- Resolution is per-row and tolerant: a session the reader can't read is simply left untitled; one unreadable session never blanks the rest.
- The desktop layer copies `sessionTitle` → `sessionName`; the renderer (`usageSessionDisplayLabel`) already owns the untitled fallback, so no UI change.

Fixes #4218

## Review responses

This supersedes the earlier main-process/catalog-join approach. Addressing the review:

- **Spec P2 (catalog is not a complete usage-session source).** Confirmed: `readCatalogRecord` applies the same `role`/`preparing`/`ledger-v0` filters as the list query, so *no* complete name source is reachable from the desktop process. Fixed at the source — the Host resolves titles from the unfiltered durable header, covering reserved-role/coordination/legacy sessions.
- **Spec P2 (one legacy session blanks all titles).** Gone: resolution is per-row via `readHeaderSnapshot`, wrapped per-id, so a single unreadable/unprojectable session can't affect the others. The all-or-nothing `client.listSessions()` is no longer used here.
- **Standards P2 (reconnect exception swallowed).** Gone: the separate `loadSessionNames()` catch was removed. The title now rides the existing `queryUsage` read, which already propagates reconnectable failures through `handleReconnectableRead`.
- **Standards P3 (blank-title test not actually covered).** Covered on the Host: a coordinator test seeds a whitespace-only title and asserts the row is emitted *without* `sessionTitle` (removing the `.trim()` guard fails it), alongside a resolved title and a throwing (unreadable) session.

## Verification

Same Storybook case (`product-settings-pages--usage-long-tail`), differing only by whether the title is populated. Named sessions show their real title; genuinely unnamed rows still fall back:

Usage 任务 column before/after

Checks run locally:

- `tsc` build of `@maka/runtime-host` and `tsc -p tsconfig.main.json` (desktop main) — both clean.
- `node --test` — 23/23 across the affected suites:
  - `usage-pricing-two-client-uds` incl. a new case: the LLM/tool logs carry the Host-resolved title, a whitespace-only title is dropped, and an unreadable session is tolerated.
  - `usage-pricing-protocol` incl. `sessionTitle` round-trip + over-length rejection.
  - `runtime-host-usage-ipc-main`: `sessionName` is surfaced from the projection's `sessionTitle`.
- `biome check` on the changed files — clean.

Not run in this environment: the full workspace typecheck / full test suite / e2e / a real Electron window. `@maka/ui` currently has pre-existing `@astryxdesign/core` type errors unrelated to this change; the desktop **main** process and the Host packages don't depend on `@maka/ui`, and Storybook renders it from source, so the evidence above is unaffected.

## AI use

Select exactly one:

- [ ] No generative tool made a substantive contribution
- [x] Generative tooling made a substantive contribution

Tool(s) and scope: Claude Code (claude-opus-4-8) — diagnosis, the fix, the tests, and the before/after evidence. The commit carries a `Generated-by` trailer.

## Checklist

- [x] Tests cover the change and fail without it
- [x] Lint, format, typecheck and the affected suites pass locally

Does this PR entail a change in behavior?

- [x] Yes — described under Summary above
- [ ] No

### CODE REVIEW

File: packages/runtime-host/src/server/usage-pricing-coordinator.ts [hunk 1]
```diff
@@ -75,18 +75,49 @@ export class HostUsagePricingCoordinator {
...
+    if (!read) return titles;
+    const ids = [
+      ...new Set(rows.map((row) => row.sessionId).filter((id): id is string => id !== undefined)),
+    ];
+    await Promise.all(
+      ids.map(async (id) => {
+        try {
+          const title = (await read(id))?.trim();
+          if (title) titles.set(id, title);
+        } catch {
```
Comment (Astro-Han) [inline packages/runtime-host/src/server/usage-pricing-coordinator.ts]: [P2] Suggestion (category ② — reasonable storage failure/recovery): could we catch only the explicit Session-not-found case here? `readHeaderSnapshot()` can also fail because the metadata store is closing, SQLite I/O failed, or a durable header is malformed. Swallowing every error makes `usage.query` return success with an untitled label, so the outer `#mapReadFailure()` never emits `host_draining`/`persistence_failed` and the Desktop loses its normal reconnect path. I suggest tolerating `isSessionNotFoundError(error)` while rethrowing lifecycle/persistence/invalid-document failures, with one missing-Session case and one propagated storage-failure regression. Please push back if all title reads are deliberately specified as non-authoritative best effort, including store lifecycle failures.

**Review by Astro-Han — COMMENTED** [review]
Thanks for restoring durable Session titles in the Usage Activity projection. I reviewed exact head `5e2c85c15ce380799dbf8a5808794ca6daaa6e42`; the problem in #4218 is demonstrated, the Runtime Host is the right owner, both LLM and tool rows are covered, and the exact-head checks are green. I left one P2 recovery-path suggestion inline. This is an adjacent reviewer’s reading, so please do push back if the title-reader failure contract is intentionally best-effort for all error classes.

<details><summary>中文摘要</summary>

谢谢把 Session 标题恢复到 Usage Activity。问题、owner 和主路径都成立，当前 exact-head checks 也通过；唯一建议是不要把标题读取的所有存储异常都静默降级为“未命名”，否则会绕过现有 draining/persistence/reconnect 恢复路径。若这里的合同明确要求所有错误均 best-effort，欢迎直接 push back。

</details>

AI-assisted review disclosure: Codex ran an independent analysis lane; Astro-Han independently verified the exact head, production path, and severity, and owns this review.

File: packages/runtime-host/src/server/usage-pricing-coordinator.ts (hunk 1 above)
Comment (liuxiaocs7) [reply packages/runtime-host/src/server/usage-pricing-coordinator.ts]: Applied — thanks, this is the right call. The title read is best-effort **only for a genuinely missing session**, not for store failures.

`#resolveSessionTitles` now tolerates just `isSessionNotFoundError(error)` and rethrows everything else, so a draining/closed store, SQLite I/O, or a malformed durable header propagates through `#queryUsage` → `#mapReadFailure()` and surfaces as `host_draining` / `persistence_failed` — the Desktop keeps its normal reconnect path instead of a spurious untitled row.

Covered both ways in `usage-pricing-two-client-uds`:
- the existing missing-session row now throws a real `SessionNotFoundError` and stays tolerated (untitled), and
- a new case whose reader throws `InteractiveUsageStoresClosedError` asserts `usage.query` fails with `host_draining` (dropping the `isSessionNotFoundError` guard fails it).

Pushed as `d77975feb` (also rebased onto latest main).

File: packages/runtime-host/src/__tests__/usage-pricing-two-client-uds.test.ts [hunk 2]
```diff
@@ -295,6 +299,105 @@ test('pricing root identity failure requests drain while expected failures do no
...
+    const coordinator = new HostUsagePricingCoordinator(
+      stores,
+      () => {},
+      new RuntimePolicyActivationGate(),
+      () => {},
+      // Not a missing session — the metadata store is draining/closed. This must
+      // not be swallowed into an untitled row; it has to reach the Desktop's
+      // normal reconnect path via #mapReadFailure.
+      async () => {
+        throw new InteractiveUsageStoresClosedError();
```
Comment (Astro-Han) [inline packages/runtime-host/src/__tests__/usage-pricing-two-client-uds.test.ts]: Thanks for tightening the catch to the real not-found case. [P3] (category ② — storage failure/recovery) This regression injects `InteractiveUsageStoresClosedError` into the session-title callback, but the production callback is `SessionStore.readHeaderSnapshot`; it cannot emit a usage-store lifecycle error. The test therefore proves the coordinator classifier with a fabricated cross-store error, not that an actual session-metadata failure reaches the reconnectable path. Could we replace this case, rather than add another test, with the real execution-store/SessionStore failure at the composition boundary, or narrow the test name and claim to what it actually covers? Please push back if there is an adapter that translates production title-read failures to this exact type.

**Review by Astro-Han — COMMENTED** [review]
Thanks for the follow-up. I rechecked the exact head, and the original error-swallowing P2 is fixed. I have one test-quality suggestion below. This was an AI-assisted review using Codex; I independently checked the production wiring and exact-head CI. Please push back if I have missed an invariant in the storage composition.

**Review by Astro-Han — APPROVED** [review]
Thanks for the quick test cleanup. I rechecked exact head e415a1aff: the production title lookup now tolerates only a genuinely missing Session, propagates other metadata failures, and the revised regression no longer claims a fabricated usage-store lifecycle mapping. This was an AI-assisted review using Codex; I independently verified the production composition and exact delta and own this approval. Please push back if any storage invariant has been misread.

File: packages/runtime-host/src/__tests__/usage-pricing-two-client-uds.test.ts (hunk 2 above)
Comment (liuxiaocs7) [reply packages/runtime-host/src/__tests__/usage-pricing-two-client-uds.test.ts]: Good catch — you're right, and there's no such adapter. `SessionStore.readHeaderSnapshot`'s only usage-relevant failures are `SessionNotFoundError` (tolerated) or, for a closed metadata store, a plain `Error('SQLite session metadata store is closed')`. That's `unknown` to `classifyInteractiveUsageStoresFailure`, so `#mapReadFailure` rethrows it — it never becomes `host_draining`. My `InteractiveUsageStoresClosedError` case was asserting a cross-store mapping the production reader can't reach.

Replaced (not added) that case in `e415a1aff`: it now injects the exact generic error a closed metadata store throws and asserts `usage.query` **propagates/rejects** rather than returning a false-success untitled page, with the name + comment narrowed to that claim (no `host_draining`).

On the reconnect path: a genuinely draining host is already caught by the *primary* usage read (`telemetry.logs`/canonical), which runs before title resolution and emits `host_draining` itself. The title read only reaches this branch when the session store fails on its own — there the fix's job is just to surface it, not mask it, which the narrowed test proves. Missing-session tolerance is unchanged.
